integration/build_test.go

```diff
@@ -32,96 +31,80 @@ import (
+	if err != nil {
+		t.Fatal(err)
+	}
 
-		startNamespace := getCurrentNamespace()
-		defer changeToNamespace(ctx, oktetoPath, startNamespace)
+	pathToManifestDir := filepath.Join(cwd, repoDir)
+	if err := writeFile(pathToManifestDir, manifestFilename, manifestContent); err != nil {
+		t.Fatal(err)
+	}
+	pathToManifestFile := filepath.Join(pathToManifestDir, manifestFilename)
 
-		if err := createNamespace(ctx, oktetoPath, namespace); err != nil {
-			t.Fatal(err)
-		}
-		defer deleteNamespace(ctx, oktetoPath, namespace)
+	t.Cleanup(func() {
+		changeToNamespace(ctx, oktetoPath, originNamespace)
+		deleteNamespace(ctx, oktetoPath, testNamespace)
+		deleteGitRepo(ctx, repoDir)
 
-		if err := cloneGitRepo(ctx, buildGitRepo); err != nil {
-			t.Fatal(err)
-		}
-		defer deleteGitRepo(ctx, buildGitFolder)
+	})
 
-		cwd, err := os.Getwd()
-		if err != nil {
-			t.Fatal(err)
-		}
+	t.Run("okteto build should build and push the image to registry", func(t *testing.T) {
 
-		dir, err := os.MkdirTemp(cwd, testName)
-		if err != nil {
-			t.Fatal(err)
-		}
-		defer os.RemoveAll(dir)
-
-		manifestPath := filepath.Join(dir, "okteto.yml")
-		manifestContent := `build:
-  react-getting-started:
-    context: .
-`
-		if err := writeManifestV2(manifestPath, manifestContent); err != nil {
-			t.Fatal(err)
+		if _, err := registry.GetImageTagWithDigest(expectedImageTag); err == nil {
+			t.Fatal("image is already at registry")
 		}
 
-		t.Logf("running okteto build command...")
-		if err := oktetoBuild(ctx, oktetoPath, manifestPath); err != nil {
+		_, err := runOktetoBuild(ctx, oktetoPath, pathToManifestFile, repoDir)
+		if err != nil {
 			t.Fatal(err)
 		}
 
-		t.Logf("checking at registry if the image has been built...")
-		expectedImageTag := fmt.Sprintf("%s/%s/%s:dev", okteto.Context().Registry, namespace, buildGitFolder)
 		if _, err := registry.GetImageTagWithDigest(expectedImageTag); err != nil {
-			t.Fatal(err)
+			t.Fatalf("image not pushed to registry: %s", err.Error())
 		}
 
 	})
 }
 
-func oktetoBuild(ctx context.Context, oktetoPath, oktetoManifestPath string) error {
-	cmd := exec.Command(oktetoPath, "build", "-f", oktetoManifestPath)
+func runOktetoBuild(ctx context.Context, oktetoPath, pathToManifestFile, repoDir string) (string, error) {
+	cmd := exec.Command(oktetoPath, "build", "-f", pathToManifestFile, "-l", "debug")
 	cmd.Env = append(os.Environ(), "OKTETO_ENABLE_MANIFEST_V2=true")
-	cmd.Dir = buildGitFolder
+	cmd.Dir = repoDir
 	o, err := cmd.CombinedOutput()
 	if err != nil {
-		return fmt.Errorf("okteto build failed: %s - %s", string(o), err)
-	}
-	log.Printf("okteto build %s success", oktetoManifestPath)
-	return nil
-}
-
-func writeManifestV2(path, content string) error {
-	oFile, err := os.Create(path)
-	if err != nil {
-		return err
-	}
-	defer oFile.Close()
-
-	if err := os.WriteFile(oFile.Name(), []byte(content), 0600); err != nil {
-		return err
+		return "", fmt.Errorf("okteto build failed: %s - %s", string(o), err)
```

Comment:
Can we show the output of the command before returning?. This way it's easier to debug errors on CI